core/src/main/java/org/jdbi/v3/core/generic/internal/Preconditions.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.jdbi.v3.core.generic.internal;
+
+import javax.annotation.Nullable;
+
+public final class Preconditions {
+  private Preconditions() {}
+
+  public static void checkArgument(boolean expression) {
+    if (!expression) {
+      throw new IllegalArgumentException();
+    }
+  }
+
+  public static void checkArgument(
+      boolean expression,
+      @Nullable String errorMessageTemplate,
+      @Nullable Object... errorMessageArgs) {
+    if (!expression) {
+      throw new IllegalArgumentException(format(errorMessageTemplate, errorMessageArgs));
+    }
+  }
+
+  public static <T> T checkNotNull(T reference) {
+    if (reference == null) {
+      throw new NullPointerException();
+    }
+    return reference;
+  }
+
+  public static void checkState(boolean expression) {
+    if (!expression) {
+      throw new IllegalStateException();
+    }
+  }
+
+  public static void checkState(
+          boolean expression,
+          @Nullable String errorMessageTemplate,
+          @Nullable Object... errorMessageArgs) {
+    if (!expression) {
+      throw new IllegalStateException(format(errorMessageTemplate, errorMessageArgs));
+    }
+  }
+
+  static String format(String template, @Nullable Object... args) {
```

Comment:
Is this so that arguments get appended to the end if you forget a format specifier?  The implementation looks really strange to me, it is called `format` like `String.format` but if you try to use *any* documented format specifier other than `%s` it will silently behave wrong.  I'm not sure that surprise is worth the nicety of having forgotten arguments appended.